src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Methods to parse strings typed by user.
+ * 
+ * @author Tianqi-Zhu
+ */
+public class Parser {
+    
+    /**
+     * Parse a line of texts typed by user. 
+     * 
+     * @param newInput User's new line of input. 
+     * @param taskList Current taskList.
+     * @return Commands to execute or tasks to put into list.
+     * @throws DukeExcpetion If the input is valid.
+     */
+    public static Executable parse(String newInput, TaskList taskList) throws DukeExcpetion {
```

Comment:
Perhaps the if-else conditions can be separated into different methods? I think this will greatly increase the readability of your code as the parse method is currently quite verbose.